File: 9-palindrome-number/palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        String str = Integer.toString(x);
        char[] char1 = str.toCharArray();
        char[] y = new char[char1.length];
        for (int i = 0; i < char1.length; i++) {
            y[i] = char1[char1.length - 1 - i];
        }
        if (Arrays.equals(char1, y)) {
            return true;
        } else {
            return false;
        }
    }
}

File: 9-palindrome-number/palindrome-number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        int dup = x;
        int rev = 0;
        while(x>0){
            int ld = x % 10;
            x = x/10;
            if((rev > INT_MAX/10) || (rev < INT_MIN/10)) return 0;
            rev = (rev * 10) + ld;
        }
        if(rev == dup) return true;
        else return false;
    }
};